adafruit_fram.py

```diff
@@ -0,0 +1,273 @@
+            self._wp_pin.switch_to_output()
+            self._wp_pin.value = self._wp
+        else:
+            self._wp_pin = wp_pin
+
+    @property
+    def max_size(self):
+        """ The maximum size of the current FRAM chip. This is the highest
+            register location that can be read or written to.
+        """
+        return self._max_size
+
+    @property
+    def write_wraparound(self):
+        """ Determines if sequential writes will wrapaound the ``FRAM.max_size``
+            address. If ``False``, and a requested write will extend beyond the
+            maximum size, an exception is raised.
+        """
+        return self._wraparound
+
+    @write_wraparound.setter
+    def write_wraparound(self, value):
+        if not value in (True, False):
+            raise ValueError("Write wraparound must be 'True' or 'False'.")
+        self._wraparound = value
+
+    @property
+    def write_protected(self):
+        """ The status of write protection. Default value on initialization is
+            ``False``.
+
+            When a ``WP`` pin is supplied during initialization, or using
+            ``write_protect_pin``, the status is tied to that pin and enables
+            hardware-level protection.
+
+            When no ``WP`` pin is supplied, protection is only at the software
+            level in this library.
+        """
+        return self._wp if self._wp_pin is None else self._wp_pin.value
+
+    @write_protected.setter
+    def write_protected(self, value):
+        self._wp = value
+        if not self._wp_pin is None:
+            self._wp_pin.value = value
+
+    def __getitem__(self, key):
+        """ Read the value at the given index, or values in a slice.
+
+            .. code-block:: python
+
+                # read single index
+                fram[0]
+
+                # read values 0 thru 9 with a slice
+                fram[0:9]
+
+                # read every other value from 0 thru 10 using a step
+                fram[0:10:2]
+        """
+        if isinstance(key, int):
+            if key > self._max_size:
+                raise ValueError("Register '{0}' greater than maximum FRAM size."
+                                 " ({1})".format(key, self._max_size))
+            read_buffer = self._read_byte(key)
+        elif isinstance(key, slice):
+            registers = list(range(key.start if not key.start is None else 0,
+                                   key.stop if not key.stop is None else self._max_size,
+                                   key.step if not key.step is None else 1))
+            if (registers[0] + len(registers)) > self._max_size:
+                raise ValueError("Register + Length greater than maximum FRAM size."
+                                 " ({0})".format(self._max_size))
+
+            read_buffer = bytearray(len(registers))
+            for i, register in enumerate(registers):
+                read_buffer[i] = self._read_byte(register)[0]
```

Comment:
Is it possible to read more than one byte at a time? That will be much faster.

Reply:
At this point, I don't remember why I chose to read one byte at a time. May have been an attempt to minimize memory use. 

Just ran some comparisons; memory allocation is tripped at the same length of reads both ways. I'll switch it. (it is much faster)